metricbeat/module/windows/perfmon/reader.go

```diff
@@ -137,33 +137,50 @@ func (r *Reader) RefreshCounterPaths() error {
 }
 
 // Read executes a query and returns those values in an event.
-func (r *Reader) Read() ([]mb.Event, error) {
+func (this *Reader) Read(metricset string) ([]mb.Event, error) {
 	// Some counters, such as rate counters, require two counter values in order to compute a displayable value. In this case we must call PdhCollectQueryData twice before calling PdhGetFormattedCounterValue.
 	// For more information, see Collecting Performance Data (https://docs.microsoft.com/en-us/windows/desktop/PerfCtrs/collecting-performance-data).
-	if err := r.query.CollectData(); err != nil {
+	if err := this.query.CollectData(); err != nil {
 		return nil, errors.Wrap(err, "failed querying counter values")
 	}
 
 	// Get the values.
-	values, err := r.query.GetFormattedCounterValues()
+	values, err := this.query.GetFormattedCounterValues()
 	if err != nil {
 		return nil, errors.Wrap(err, "failed formatting counter values")
 	}
+	var events []mb.Event
+	if metricsetName != metricset {
```

Comment:
Why this difference?

Reply:
// GroupAllCountersTo config option will only apply to the webserver metricset, where counters for all instances are aggregated, I do not plan to make this option available to the perfmon metricset yet

Reply:
removed the check for the metricset name